### DISCUSSION TITLE
Terrible generation slowdown on resolutions larger than 512x512

### DISCUSSION THREAD

**Discussion by ostap667inbox** [vladmandic/sdnext#1267] [Q&A]
After the last updates () got a catastrophic drop in performance. Generation of 512x512 images goes about the same speed as before. However, increasing resolution to at least 768x768 leads to a wild slowdown. Look at the screenshot - first line - generation 512x512, second - 768x768. Only the resolution has changed. 
![chrome_M2olEVOo3i](https://github.com/vladmandic/automatic/assets/123593542/a5b8379a-c99d-4da2-8995-345da8a5828a)

No scripts, ControlNet, Hires,fix etc. were used - only the request.
Tried different sampling method, all the same. 

At the same time generation often hangs for a long time on start. In resources monitor I see almost full VRAM load (12Gb RTX 3060). Switched Scaled-Dot-Product, Doggettx's, turned everything off completely. I don't understand what's going on. There are no error messages of any kind. Any idea what's going on, what can be done in this situation?

**Comment by vladmandic** [comment]
there is nothing in recent updates that would have any impact on this. plus i've tried scaling resolution and performance is pretty much linear with number of pixels.

**Reply by ostap667inbox** [reply]
The problem seems to be solved by the `--reinstall` option in `webui-user.bat`. 
The VRAM leakage is gone and the generation speed is now proportional to the resolution.

**Comment by kyism** [comment]
Restore GeForce driver to 531.
DONT update to 532+.

i got this performance issue two days ago. restoring to 531 worked to me.
btw, A1111 is fine with 532, maybe the torch version is different.

**Reply by vladmandic** [reply]
i have 532 without issues.

**Reply by ghost** [reply]
I have the newer one too because I wanted to try olive, there was no change in performance, but I read that on reddit some users reported a drop after installing this driver

**Reply by vladmandic** [reply]
why is everyone crazy about olive? its tensor cores support for onnx - totally not applicable.

**Reply by kyism** [reply]
after receiving vladmandic's reply, i tested serveral scenarioes. i got the workaround. 532/535 indeed chabge the VRAM management. i usually set batch size 8 to gen mutilple images before 532, and now i only can set 3 without performance drop. the new driver moves data to Shared GPU Memory, the old one never uses Shared GPU Memory. therefore,in 532/535,if Shared GPU Memory is allocated in windows task manager, just decrease the image size or batch size. A1111 also applies the workaround one new driver.

**Reply by ghost** [reply]
> why is everyone crazy about olive? its tensor cores support for onnx - totally not applicable.

I **was** crazy about it at first. Basically anything that provides more it/s is beeping on my radar screen 😄

**Reply by radry** [reply]
> after receiving vladmandic's reply, i tested serveral scenarioes. i got the workaround. 532/535 indeed chabge the VRAM management. i usually set batch size 8 to gen mutilple images before 532, and now i only can set 3 without performance drop. the new driver moves data to Shared GPU Memory, the old one never uses Shared GPU Memory. therefore,in 532/535,if Shared GPU Memory is allocated in windows task manager, just decrease the image size or batch size. A1111 also applies the workaround one new driver.

That makes the new drivers pretty much unusable.

**Reply by vladmandic** [reply]
@kyism 

> A1111 also applies the workaround one new driver.

 what workaround?

**Reply by kyism** [reply]
sorry for typos. i mean there also is the same performance issue when using A1111 webui with 532/535 driver. so we can decrease image size or batch size to keep performance as usual. especially the batch size is the main cause of performance dropping in my test.
